stl/inc/mdspan

```diff
@@ -348,11 +348,13 @@ public:
         if constexpr (_Extents::rank() == 0) {
             return 1;
         } else {
-            typename _Extents::index_type _Result = 1;
+            // Use an unsigned type to prevent overflow when called from mdspan::size.
+            typename _Extents::size_type _Result = 1;
             for (size_t _Dim = 0; _Dim < _Idx; ++_Dim) {
-                _Result *= _Exts.extent(_Dim);
+                _Result *= static_cast<_Extents::size_type>(_Exts.extent(_Dim));
             }
-            return _Result;
+
+            return static_cast<_Extents::index_type>(_Result);

```

Comment:
This may result in a negative number if `index_type` is signed, but I think the only way that can happen without a precondition violation is in the call from `mdspan::size`. This is a well-defined conversion from C++20 on, and casting back to `size_type`, which `size()` does, recovers the original product.

This is the reason that the specialization below, for the `rank_dynamic() == 0` case, works as-is.